src/core/lib/security/credentials/tls/grpc_tls_crl_provider.h

```diff
@@ -36,6 +36,15 @@ class CrlImpl : public Crl {
   X509_CRL* crl_;
 };
 
+class CertificateInfoImpl : public CertificateInfo {
```

Comment:
This is fine for now, but I don't think we will actually know whether this will be the right implementation until we see the actual code in gRPC that will call the `CrlProvider`.  It might very well turn out that we want a more bespoke implementation here, tied into the SSL TSI code.

Reply:
Noted, we'll see how it evolves as I write the full impl

Reply:
I'd like to see the missing plumbing here so that we know whether or not this is the right approach.  It looks like you've added the code to use the provider in the low-level TSI code; all you're really missing is the plumbing to allow users to specify the provider via TlsCredentialsOptions and then passing that down through TlsCreds to the TSI layer.  Without that plumbing, no one can actually use this feature.